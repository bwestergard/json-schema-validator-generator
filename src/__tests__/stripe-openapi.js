// @flow
import _ from 'lodash';
import path from 'path';
import generateValidator from '../generate-validator.js';
import generateFlow from '../generate-flow.js';

const MODELS_TO_SNAPSHOT = [
  'account',
  'balance_transaction',
  'card',
  'charge',
  'coupon',
  'customer',
  'discount',
  'dispute',
];
describe('Stripe openapi', () => {
  // $FlowFixMe
  const spec = require(path.join(__dirname, '../../', 'openapi/openapi/spec2.json'));
  it('generates a validator for the entire spec', () => {
    const definitions = _.pick(spec.definitions, MODELS_TO_SNAPSHOT);
    expect(generateValidator(spec, definitions)).toMatchSnapshot();
  });
  MODELS_TO_SNAPSHOT.forEach((model) => {
    it(`generates flow types for ${model}`, () => {
      expect(generateFlow(spec.definitions[model])).toMatchSnapshot();
    });
  });
});
