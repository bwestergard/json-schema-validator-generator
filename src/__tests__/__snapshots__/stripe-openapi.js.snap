// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stripe openapi generates a validator for the entire spec 1`] = `
"function f0(v0) {
  return null;
}
return f0;"
`;

exports[`Stripe openapi generates flow types for account 1`] = `
"{
  active_payment_methods: Array<mixed>,
  business_name: ?string,
  business_url: ?string,
  charges_enabled: boolean,
  country: string,
  debit_negative_balances: boolean,
  decline_charge_on: mixed,
  default_currency: string,
  details_submitted: boolean,
  display_name: ?string,
  email: ?string,
  external_accounts: {
    data: Array<{[key: string]: mixed}>,
    has_more: boolean,
    object: \\"list\\",
    total_count: ?number,
    url: string,
  },
  id: string,
  legal_entity: mixed,
  managed: boolean,
  mcc: ?string,
  metadata: ?{[key: string]: mixed},
  object: string,
  orders: mixed,
  payout_schedule: mixed,
  payout_statement_descriptor: ?string,
  payouts_enabled: boolean,
  product_description: ?string,
  statement_descriptor: ?string,
  support_email: ?string,
  support_phone: ?string,
  timezone: ?string,
  tos_acceptance: mixed,
  verification: mixed,
}"
`;

exports[`Stripe openapi generates flow types for balance_transaction 1`] = `
"{
  amount: number,
  available_on: number,
  created: number,
  currency: string,
  description: ?string,
  fee: number,
  fee_details: mixed,
  id: string,
  net: number,
  object: string,
  source: ?string,
  status: string,
  type: string,
}"
`;

exports[`Stripe openapi generates flow types for card 1`] = `
"{
  account: ?string,
  address_city: ?string,
  address_country: ?string,
  address_line1: ?string,
  address_line1_check: ?string,
  address_line2: ?string,
  address_state: ?string,
  address_zip: ?string,
  address_zip_check: ?string,
  available_payout_methods: ?Array<mixed>,
  brand: string,
  country: ?string,
  currency: ?string,
  customer: ?string,
  cvc_check: ?string,
  default_for_currency: ?boolean,
  dynamic_last4: ?string,
  estimated_availability: ?string,
  exp_month: number,
  exp_year: number,
  fingerprint: ?string,
  funding: string,
  google_reference: ?string,
  id: string,
  last4: string,
  metadata: {[key: string]: mixed},
  name: ?string,
  object: string,
  recipient: ?string,
  three_d_secure: ?{[key: string]: mixed},
  tokenization_method: ?string,
}"
`;

exports[`Stripe openapi generates flow types for charge 1`] = `
"{
  amount: number,
  amount_authorized: ?number,
  amount_captured: number,
  amount_refunded: number,
  application: ?string,
  application_fee: ?string,
  balance_transaction: ?string,
  captured: boolean,
  card: mixed,
  created: number,
  currency: string,
  customer: ?string,
  description: ?string,
  destination: ?string,
  dispute: ?string,
  failure_code: ?string,
  failure_message: ?string,
  fraud_details: ?{[key: string]: mixed},
  id: string,
  invoice: ?string,
  livemode: boolean,
  metadata: {[key: string]: mixed},
  object: string,
  on_behalf_of: ?string,
  order: ?string,
  outcome: mixed,
  paid: boolean,
  receipt_email: ?string,
  receipt_number: ?string,
  refunded: boolean,
  refunds: {
    data: Array<mixed>,
    has_more: boolean,
    object: \\"list\\",
    total_count: ?number,
    url: string,
  },
  review: ?string,
  shipping: mixed,
  source_transfer: ?string,
  statement_descriptor: ?string,
  status: string,
  transfer: ?string,
  transfer_group: ?string,
}"
`;

exports[`Stripe openapi generates flow types for coupon 1`] = `
"{
  amount_off: ?number,
  created: number,
  currency: ?string,
  duration: string,
  duration_in_months: ?number,
  id: string,
  livemode: boolean,
  max_redemptions: ?number,
  metadata: {[key: string]: mixed},
  object: string,
  percent_off: ?number,
  redeem_by: ?number,
  times_redeemed: number,
  valid: boolean,
}"
`;

exports[`Stripe openapi generates flow types for customer 1`] = `
"{
  account_balance: number,
  alipay_accounts: ?{
    data: Array<mixed>,
    has_more: boolean,
    object: \\"list\\",
    total_count: ?number,
    url: string,
  },
  bank_accounts: ?{
    data: Array<mixed>,
    has_more: boolean,
    object: \\"list\\",
    total_count: ?number,
    url: string,
  },
  business_vat_id: ?string,
  cards: {
    data: Array<mixed>,
    has_more: boolean,
    object: \\"list\\",
    total_count: ?number,
    url: string,
  },
  created: number,
  currency: ?string,
  default_bank_account: ?string,
  default_card: ?string,
  default_source: ?string,
  delinquent: ?boolean,
  description: ?string,
  discount: mixed,
  email: ?string,
  id: string,
  livemode: boolean,
  metadata: {[key: string]: mixed},
  object: string,
  shipping: mixed,
  sources: {
    data: Array<{[key: string]: mixed}>,
    has_more: boolean,
    object: \\"list\\",
    total_count: ?number,
    url: string,
  },
  subscriptions: {
    data: Array<mixed>,
    has_more: boolean,
    object: \\"list\\",
    total_count: ?number,
    url: string,
  },
}"
`;

exports[`Stripe openapi generates flow types for discount 1`] = `
"{
  coupon: mixed,
  customer: ?string,
  end: ?number,
  object: string,
  start: number,
  subscription: ?string,
}"
`;

exports[`Stripe openapi generates flow types for dispute 1`] = `
"{
  amount: number,
  balance_transactions: mixed,
  charge: string,
  created: number,
  currency: string,
  evidence: {[key: string]: mixed},
  evidence_details: {[key: string]: mixed},
  id: string,
  is_charge_refundable: boolean,
  livemode: boolean,
  metadata: {[key: string]: mixed},
  object: string,
  reason: string,
  status: string,
}"
`;
